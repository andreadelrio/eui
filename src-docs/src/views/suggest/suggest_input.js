import React, { Component } from 'react';

import {
  EuiButtonEmpty,
  EuiIcon,
  EuiSpacer,
  EuiSuggestInput,
} from '../../../../src/components';

const statusList = ['notYetSaved', 'saved', 'noNewChanges', 'isLoading'];

export default class extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: '',
    };
  }

  onChange = e => {
    this.setState({
      value: e.target.value,
    });
  };

  render() {
    const hashtag = (
      <EuiButtonEmpty iconType="arrowDown" iconSide="right">
        <EuiIcon type="number" />
      </EuiButtonEmpty>
    );

    return (
      <div>
        {statusList.map((item, index) => (
          <div>
            <EuiSuggestInput
              onChange={this.onChange}
              value={this.state.value}
              key={index}
              status={item}
              action={hashtag}
              label={'KQL'}
            />
            <EuiSpacer size="m" />
          </div>
        ))}
      </div>
    );
  }
}
