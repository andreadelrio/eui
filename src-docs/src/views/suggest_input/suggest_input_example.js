import React from 'react';

import { renderToHtml } from '../../services';

import {
  GuideSectionTypes,
} from '../../components';

import {
  EuiCode,
  EuiSuggestInput,
} from '../../../../src/components';

import SuggestInput from './suggest_input';
const suggestInputSource = require('!!raw-loader!./suggest_input');
const suggestInputHtml = renderToHtml(SuggestInput);

export const SuggestInputExample = {
  title: 'SuggestInput',
  sections: [{
    title: 'SuggestInput',
    source: [{
      type: GuideSectionTypes.JS,
      code: suggestInputSource,
    }, {
      type: GuideSectionTypes.HTML,
      code: suggestInputHtml,
    }],
    text: (
      <p>
        Description needed: how to use the <EuiCode>EuiSuggestInput</EuiCode> component.
      </p>
    ),
    props: { EuiSuggestInput },
    demo: <SuggestInput />,
  }],
};
