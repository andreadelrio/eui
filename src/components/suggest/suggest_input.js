/* eslint-disable @typescript-eslint/no-unused-vars */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { EuiFilterButton } from '../filter_group';
import { EuiFieldText } from '../form';
import { EuiToolTip } from '../tool_tip';
import { EuiIcon } from '../icon';
import { EuiPopover, EuiInputPopover } from '../popover';

export class EuiSuggestInput extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: '',
      isPopoverOpen: false,
    };
  }

  onFieldChange(e) {
    this.setState({
      value: e.target.value,
      isPopoverOpen: e.target.value !== '' ? true : false,
    });
    this.props.sendValue(e.target.value);
  }

  closePopover() {
    this.setState({
      isPopoverOpen: false,
    });
  }

  render() {
    const {
      className,
      status,
      append,
      prefix,
      tooltipContent,
      suggestions,
      sendValue,
      ...rest
    } = this.props;

    const statusMap = {
      unsaved: {
        icon: 'dot',
        color: 'accent',
      },
      saved: {
        icon: 'checkInCircleFilled',
        color: 'secondary',
      },
      unchanged: {
        icon: '',
        color: 'secondary',
      },
    };

    let icon;
    let color;
    let tooltip;

    if (statusMap[status]) {
      icon = statusMap[status].icon;
      color = statusMap[status].color;
    }
    const classes = classNames('euiSuggestInput', className);

    // EuiFieldText's append accepts an array of elements so start by creating an empty array
    const appendArray = [];

    const statusElement = status !== 'isLoading' && (
      <EuiToolTip
        position="left"
        content={tooltipContent}
        anchorClassName="euiSuggestInput__statusIcon">
        <EuiIcon color={color} type={icon} />
      </EuiToolTip>
    );

    // Push the status element to the array if it is not undefined
    if (statusElement) appendArray.push(statusElement);

    // Check to see if consumer passed an append item and if so, add it to the array
    if (append) appendArray.push(append);

    const customInput = (
      <EuiFieldText
        value={this.state.value}
        fullWidth
        prepend={prefix}
        append={appendArray}
        isLoading={status === 'isLoading' ? true : false}
        onChange={this.onFieldChange.bind(this)}
        {...rest}
      />
    );

    return (
      <div className={classes}>
        <EuiInputPopover
          id="popover"
          input={customInput}
          isOpen={this.state.isPopoverOpen}
          panelPaddingSize="none"
          closePopover={this.closePopover.bind(this)}>
          <div>{suggestions}</div>
        </EuiInputPopover>
      </div>
    );
  }
}

EuiSuggestInput.propTypes = {
  className: PropTypes.string,
  /**
   * Status of the current query 'unsaved', 'saved', 'unchanged' or 'isLoading'.
   */
  status: PropTypes.oneOf(['unsaved', 'saved', 'unchanged', 'isLoading']),
  tooltipContent: PropTypes.string,
  /**
   * Element to be appended to the input bar.
   */
  append: PropTypes.node,
  /**
   * Element to be prepended to the input bar.
   */
  prefix: PropTypes.node,
  /**
   * List of suggestions to display using 'suggestItem'.
   */
  suggestions: PropTypes.array,
};

EuiSuggestInput.defaultProps = {
  status: 'unchanged',
};
