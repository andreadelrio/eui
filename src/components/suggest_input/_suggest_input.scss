.euiSuggestInput {
  .status {
    font-size: $euiFontSizeS;
    padding-right: $euiSizeS;
    color: $euiColorPrimary;
      .statusLabel {
        padding-left: $euiSizeS;
      }
  }
}
