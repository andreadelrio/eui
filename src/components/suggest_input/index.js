export {
  EuiSuggestInput,
} from './suggest_input';
