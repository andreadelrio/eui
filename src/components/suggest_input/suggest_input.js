/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { EuiFilterButton } from '../filter_group';
import { EuiFieldText } from '../form';
import { EuiToolTip } from '../tool_tip';
import { EuiIcon } from '../icon';

export const EuiSuggestInput = ({ className, status, value, ...rest }) => {

  const filterTriggerButton = (
    <EuiFilterButton title="title">Filters</EuiFilterButton>
  );

  const statusElement = (
    <EuiToolTip position="left" content={status.tooltip}>
      <div className="status">
        <EuiIcon color={status.color} type={status.icon} />
        <span className="statusLabel">{status.label}</span>
      </div>
    </EuiToolTip>
  );

  const classes = classNames('euiSuggestInput', className);

  return (
    <div className={classes} {...rest}>
      <EuiFieldText
        value={value}
        fullWidth
        prepend={filterTriggerButton}
        append={statusElement}
      />
    </div>
  );
};

EuiSuggestInput.propTypes = {
  className: PropTypes.string,
  /**
   * Takes 'icon' for EuiIcon, 'color'. 'color' can be either our palette colors (primary, secondary, etc) or a hex value.
   */
  status: PropTypes.object,
};

EuiSuggestInput.defaultProps = {};
