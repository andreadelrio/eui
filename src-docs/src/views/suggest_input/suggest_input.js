import React from 'react';

import { EuiSpacer, EuiSuggestInput } from '../../../../src/components';

const shortDescription = 'This is the description';

const sampleItems = [
  {
    status: {
      icon: 'dot',
      name: 'Unsaved changes',
      label: 'KQL',
      color: '#DD0A73',
      tooltip: "You've made changes to this saved query. Click to save them.",
    },
    value:
      'products.category : “tanktop” and geoip.region : “australia” and price > 5',
  },
  {
    status: {
      icon: 'checkInCircleFilled',
      name: 'Changes just saved',
      label: 'KQL',
      color: 'secondary',
      tooltip: '',
    },
    value:
      'products.category : “tanktop” and geoip.region : “australia” and price > 5',
  },
  {
    status: {
      icon: '',
      name: 'All changes saved',
      label: 'KQL',
      color: 'secondary',
      tooltip: '',
    },
    value:
      'products.category : “tanktop” and geoip.region : “australia” and price > 5',
  },
];

export default () => (
  <div>
    {sampleItems.map(item => (
      <div>
        <EuiSuggestInput status={item.status} value={item.value} />
        <EuiSpacer size="m" />
      </div>
    ))}
  </div>
);
