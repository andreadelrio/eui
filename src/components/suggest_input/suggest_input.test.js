import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test/required_props';

import { EuiSuggestInput } from './suggest_input';

describe('EuiSuggestInput', () => {
  test('is rendered', () => {
    const component = render(
      <EuiSuggestInput {...requiredProps} />
    );

    expect(component)
      .toMatchSnapshot();
  });
});
