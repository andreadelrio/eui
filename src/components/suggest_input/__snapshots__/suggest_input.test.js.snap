// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiSuggestInput is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiSuggestInput testClass1 testClass2"
  data-test-subj="test subject string"
>
  <div
    class="euiFormControlLayout euiFormControlLayout--fullWidth euiFormControlLayout--group"
  >
    <div
      class="euiPopover euiPopover--anchorDownLeft euiFormControlLayout__prepend"
      id="popover"
    >
      <div
        class="euiPopover__anchor"
      >
        <button
          class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--iconRight euiFilterButton"
          title="title"
          type="button"
        >
          <span
            class="euiButtonEmpty__content"
          >
            <span
              class="euiButtonEmpty__text"
            >
              <span
                class="euiFilterButton__textShift"
              />
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      class="euiFormControlLayout__childrenWrapper"
    >
      <input
        class="euiFieldText euiFieldText--fullWidth euiFieldText--inGroup"
        type="text"
      />
    </div>
    <span
      class="euiToolTipAnchor"
    >
      <div
        class="statusIcon"
      >
        <svg
          class="euiIcon euiIcon--medium euiIcon-isLoading"
          focusable="false"
          height="16"
          style="fill:#DD0A73"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        />
        <span
          class="statusLabel"
        />
      </div>
    </span>
  </div>
</div>
`;
